src/main/java/br/com/ead/home/models/Shift.java

```diff
@@ -0,0 +1,67 @@
+package br.com.ead.home.models;
+
+import br.com.ead.home.models.api.TimeSlot;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.SetUtils;
+
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+public record Shift(Set<TimeSlot> timeSlots) {
```

Comment:
I need to better define what is a shift!

Should a shift has multiple TimeSlots?
Should a shift be a single decorator that encapsulates the TimeSlot and represents the time a clinician can accept booking meetings?